framework/src/main/java/com/jfcbuilder/builders/XYShapeBuilder.java

```diff
@@ -0,0 +1,145 @@
+ * version 2.1 of the License, or (at your option) any later version.
+ * 
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ * 
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package com.jfcbuilder.builders;
+
+import java.awt.BasicStroke;
+import java.awt.Color;
+import java.awt.Paint;
+import java.awt.Shape;
+import java.awt.Stroke;
+
+import org.jfree.chart.annotations.XYAnnotation;
+import org.jfree.chart.annotations.XYShapeAnnotation;
+
+/**
+ * Builder for producing {@link XYShapeAnnotation} objects.
+ */
+public class XYShapeBuilder implements IXYAnnotationBuilder<XYShapeBuilder> {
+
+  // See XYChapeAnnotation for these defaults
+  public static final Color DEFAULT_LINE_COLOR = Color.BLACK;
+  public static final Stroke DEFAULT_LINE_STYLE = new BasicStroke(1.0f);
+  public static final Color DEFAULT_FILL_COLOR = null; // None specified
+
+  private Shape shape;
+  private Stroke lineStyle;
+  private Paint lineColor;
+  private Paint fillColor;
+
+  /**
+   * Hidden constructor.
+   */
+  private XYShapeBuilder() {
+    shape = null;
+    lineStyle = DEFAULT_LINE_STYLE;
+    lineColor = DEFAULT_LINE_COLOR;
+    fillColor = DEFAULT_FILL_COLOR;
+  }
+
+  /**
+   * Factory method for obtaining new instances of this class.
+   * 
+   * @return New instance of this class
+   */
+  public static XYShapeBuilder get() {
+    return new XYShapeBuilder();
+  }
+
+  public Shape shape() {
+    return shape;
+  }
+
+  public XYShapeBuilder shape(Shape shape) {
+    this.shape = shape;
+    return this;
+  }
+
+  public Stroke lineStyle() {
+    return lineStyle;
+  }
+
+  public XYShapeBuilder lineStyle(Stroke style) {
+    lineStyle = style;
+    return this;
+  }
+
+  public Paint lineColor() {
+    return lineColor;
+  }
+
+  public XYShapeBuilder lineColor(Paint color) {
+    lineColor = color;
+    return this;
+  }
+
+  public Paint fillColor() {
+    return fillColor;
+  }
+
+  public XYShapeBuilder fillColor(Paint color) {
+    fillColor = color;
+    return this;
+  }
+
+  private void checkBuildPreconditions() throws IllegalStateException {
+    if (shape == null) {
+      throw new IllegalStateException("Shape not set");
+    }
+  }
+
+  @Override
+  public XYAnnotation build() throws IllegalStateException {
+
+    checkBuildPreconditions();
+
+    return new XYShapeAnnotation(shape, lineStyle, lineColor, fillColor);
+  }
+
+  @Override
+  public void mapXToTimeIndex(long[] timeData, int indexRangeStartIndex, int indexRangeEndIndex) {
+
+    // Gapless shape annotations are not currently supported due to multiple complexities.
```

Comment:
See also https://github.com/matoos32/jfreechart-builder/issues/49#issuecomment-1325279476 in the ticket describing the approach for `NumberAxis` ("gapless").